apps/dashboard/app/lib/smart_attributes/attributes/auto_scripts.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module SmartAttributes
+  class AttributeFactory
+    # Build this attribute object. Must specify a valid directory in opts
+    #
+    # @param opts [Hash] attribute's options
+    # @return [Attributes::AutoScripts] the attribute object
+    def self.build_auto_scripts(opts = {})
+      dir = Pathname.new(opts[:directory].to_s)
+      options = if dir.directory? && dir.readable?
+                  Dir.glob("#{dir}/*.{sh,csh,bash,slurm,sbatch,qsub}").map do |file|
+                    [File.basename(file), file]
+                  end
+                else
+                  []
+                end
```

Comment:
Let's try to make this more maintainable and do something like:

```
SUPPORTED_EXTENSIONS = %w[sh csh bash slurm sbatch qsub].freeze
private
def self.script_options_from_directory(dir)
  return [] unless dir.directory? && dir.readable?

   Dir.glob("#{dir}/*.{#{SUPPORTED_EXTENSIONS.join(',')}}").map do |file|
      [File.basename(file), file]
    end
end
```
and then just do:
```
options = script_options_from_directory(dir)

Reply:
👍